### PR TITLE
Add more types

### PR DESCRIPTION
**Pull request by yinxusen**

### CODE REVIEW

File: sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala [hunk 1]
```diff
@@ -2341,7 +2350,21 @@ class Dataset[T] private[sql](
...
+          rows.foreach { row =>
+            buf.writeBytes(row.getString(idx).getBytes())
+          }
+        case DoubleType =>
+          rows.foreach { row => buf.writeDouble(row.getDouble(idx)) }
+        case FloatType =>
+          rows.foreach { row => buf.writeFloat(row.getFloat(idx)) }
+        case BooleanType =>
+          rows.foreach { row => buf.writeBoolean(row.getBoolean(idx)) }
+      }
```
Comment (BryanCutler) [inline sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala]: should there be a default case give an unsupported type error?

File: sql/core/src/test/scala/org/apache/spark/sql/DatasetToArrowSuite.scala [hunk 2]
```diff
@@ -17,100 +17,156 @@
...
+  @transient var column3: Seq[String] = _
+
+  override def beforeAll(): Unit = {
+    super.beforeAll()
+    column1 = Seq.fill(numElements)(Random.nextInt)
+    column2 = Seq.fill(numElements)(Random.nextDouble)
+    column3 = Seq.fill(numElements)(Random.nextString(Random.nextInt(100)))
+    dataset = column1.zip(column2).zip(column3)
+      .map{ case ((c1, c2), c3) => ArrowTestClass(c1, c2, c3) }.toDS()
+  }
```
Comment (BryanCutler) [inline sql/core/src/test/scala/org/apache/spark/sql/DatasetToArrowSuite.scala:52]: minor: maybe the above could be a little simplified with
```scala
dataset = Seq.fill(numElements)(ArrowTestClass(Random.nextInt, ..).toDS()
```

**Comment by BryanCutler** [comment]
@yinxusen , just  a couple minor comments otherwise looks good.  How is the string conversion coming, think you will have something tomorrow?

File: sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala [hunk 3]
```diff
@@ -2291,6 +2292,14 @@ class Dataset[T] private[sql](
     dt match {
       case IntegerType =>
         new ArrowType.Int(8 * IntegerType.defaultSize, true)
+      case StringType =>
```
Comment (BryanCutler) [inline sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala:2397]: how about unsigned Ints?

**Comment by mariusvniekerk** [comment]
So i tried an implementation that might be correct, but my knowledge of the arrow java api is limited.


https://github.com/mariusvniekerk/spark-arrow/blob/master/src/main/scala/org/apache/spark/sql/arrow/DataframeToArrow.scala

You guys might be able to steal ideas from there?

**Comment by yinxusen** [comment]
@BryanCutler Sorry I messed up the PR because I merged with spark master. 
@mariusvniekerk Thanks for the reference, it is really helpful and more comprehensive.

I'll close this one and open a new PR.

**Pull request closed without merging**